PDEP-1 first revision (scope) (#51417)
pandas-dev/pandas: web/pandas/pdeps/0001-purpose-and-guidelines.md

- Created: 3 August 2022
- Status: Accepted

> - Discussion: [#47444](https://github.com/pandas-dev/pandas/pull/47444), 
>   [#51417](https://github.com/pandas-dev/pandas/pull/51417)
> - Author: [Marc Garcia](https://github.com/datapythonista), 

Reviewer: ```suggestion
- Discussion: [#47444](https://github.com/pandas-dev/pandas/pull/47444),
  [#51417](https://github.com/pandas-dev/pandas/pull/51417)
- Author: [Marc Garcia](https://github.com/datapythonista),
```

- Status: Accepted
- Discussion: [#47444](https://github.com/pandas-dev/pandas/pull/47444)

> - Author: [Marc Garcia](https://github.com/datapythonista)

Reviewer: It probably makes sense to list this PR in the `Discussion` header next to the original one, and having you as a co-author here. I think that will help people find the discussions and know who to contact in case of something related to the PR.

pandas, in a similar way as a Python [PEP](https://peps.python.org/pep-0001/)
or a NumPy [NEP](https://numpy.org/neps/nep-0000.html).

> Bug fixes and conceptually minor changes (e.g. adding a parameter to a function) are out of the

Reviewer 2: Does the `Revision` at the top need to be bumped to 2?

scope of PDEPs. A PDEP should be used for changes that are not immediate and not obvious, when
everybody in the pandas community needs to be aware of the possibility of an upcoming change.
Such changes require detailed documentation before being implemented and frequently lead to a
significant discussion within the community.

> PDEP are appropriate for user facing changes, internal changes and significant discussions.

Reviewer 3: "PDEPs are appropriate..."  (pluralize)

Examples of topics worth a PDEP could include substantial API changes, breaking behavior changes,
moving a module from pandas to a separate repository, a refactoring of the pandas block manager

> or a proposal of a new code of conduct. It is not always trivial to know which issue has enough

Reviewer 4: Personally, I find the "proposal for a new code of conduct" the odd one in this list. I would say that's more a topic related to community organization, for which we don't necessarily plan to use the PDEP process?

### PDEP authors

Anyone can propose a PDEP, but core members need to sponsor a proposal made by non-core
contributors. To submit a PDEP as a community member, please propose the PDEP concept on
[an issue](https://github.com/pandas-dev/pandas/issues/new/choose), and find a pandas team

> member to collaborate with. They can co-author the PDEP with you and submit it to the PDEPs

Author: I think there are a few different opinions on this, and it might be worth to see how many people prefer it this way or that before deciding to make a change.

> member to collaborate with. They can co-author the PDEP with you and submit it to the PDEPs

Reviewer 2: In addition to adding the team so non-core contributors can find someone to help sponsor the PDEP, maybe noting the Slack group might be a good way to also find a sponsor?

> member to collaborate with. They can co-author the PDEP with you and submit it to the PDEPs

Reviewer 4: On the "They can co-author the PDEP with you and submit it to the PDEPs repository" part, that sounds as it is the sponsor/co-author that needs to do that. But if the core-team member is just sponsoring (guiding through the process), I think the non-core team member can certainly open a PR themselves?

> member to collaborate with. They can co-author the PDEP with you and submit it to the PDEPs

Reviewer: Is there a particular problem we're trying to solve with the requirement for a core dev sponsor of PDEPs? Feels to me just extra bureaucracy, and core devs can already close low quality PDEPs as invalid, which would be equivalent.

> member to collaborate with. They can co-author the PDEP with you and submit it to the PDEPs

Reviewer 4: > Is there a particular problem we're trying to solve with the requirement for a core dev sponsor of PDEPs? Feels to me just extra bureaucracy, and core devs can already close low quality PDEPs as invalid, which would be equivalent.

Copying here what I posted in the non-line comment about general summary of the changes: IIRC the reason for this was two-fold: to have _some_ form of quality assurance (to not be "spammed" by many drive-by PDEPs that might not need to be a PDEP, although this might not happen quickly in practice) and to help the non-core-team submitter (the sponsor would be a core-team member who takes on the responsibility of guiding the PDEP author through the process).

The "sponsor" term is inspired on the PEP process (https://peps.python.org/pep-0001/#submitting-a-pep, _"The sponsor’s job is to provide guidance to the PEP author to help them through the logistics of the PEP process (somewhat acting like a mentor)."_), but we should probably explain that in a bit more detail in the text here as well, if we keep this.

> member to collaborate with. They can co-author the PDEP with you and submit it to the PDEPs

Reviewer 4: @lithomas1 at https://github.com/pandas-dev/pandas/pull/51417/files#r1107953261

> Also, are PDEPs moving out of the main repo? (I'd be +1 on this)

We briefly discussed this at one of the first governance meetings, and in that sub-group there was general consensus to do that. At the time I brought that up on the pandas-dev mailing list (https://mail.python.org/pipermail/pandas-dev/2022-November/001547.html), and there were no objections to it (but also no explicit approvals ;)). I am still planning to do this.

> member to collaborate with. They can co-author the PDEP with you and submit it to the PDEPs

Reviewer 5: indeed. The concept of a mentor makes more sense than using the term sponsor suggests.

> member to collaborate with. They can co-author the PDEP with you and submit it to the PDEPs

Reviewer: I still think we're fixing a problem we don't have, which doesn't seem like a good idea in general. Probably when we do have too many low quality PDEPs we will want to revisit this anyway (and I feel just closing them as invalid may make more sense).

If someone from the Arrow, Dask, scikit-learn, Intel... Wants to propose a PDEP, I don't know how requiring them to have a sponsor or a mentor is helpful. I really like the idea of offering a mentor for PDEPs coming from less experienced people. But I don't see how this is helpful (if you want to spam with a low quality PDEP, you can dl it anyway whatever we say here), and I do see how this can be harmful.

> member to collaborate with. They can co-author the PDEP with you and submit it to the PDEPs

Reviewer 4: Thoughts on Irv's proposal to rename it (sponsor -> advisor)? 

Personally, I am also fine to just leave out this addition of the sponsor mention. It's not really related to the other changes to clarifying scope etc, and we can always add it later if we want.

> member to collaborate with. They can co-author the PDEP with you and submit it to the PDEPs

Reviewer: I don't like the "sponsor required" idea, happy with "advisor provided" one or just leave this out, both are fine.

> member to collaborate with. They can co-author the PDEP with you and submit it to the PDEPs

Author: I used @Dr-Irv's suggestion for now. I'd happy to get additional feedback on this in the future - I think it's an important point to get right.

The file is a markdown file, you can use `web/pdeps/0001.md` as a reference
for the expected format.

The initial status of a PDEP will be `Status: Draft`. Once it is ready for discussion, the author(s)

> change it to `Status: Under discussion`, and the following are notified: core and triage teams

Reviewer 6: Could clarify to ask your sponsor to ping the pandas-core/triage teams, since normally people can't ping/see the teams.

Author: We discussed creating some automation for the email and tag based on the label change. But I'll clarify for now.

The file is a markdown file, you can use `web/pdeps/0001.md` as a reference
for the expected format.

The initial status of a PDEP will be `Status: Draft`. Once it is ready for discussion, the author(s)
change it to `Status: Under discussion`, and the following are notified: core and triage teams 
(by tagging them on GitHub: `@pandas-core`, `@triage teams`) and the 
[pandas-dev mailing list](mailto:pandas-dev@python.org). This will be changed to `Status: 

> Accepted` when the PDEP is ready and have the approval of the core team.

Reviewer 3: Based on discussion in governance meeting on 3/15/2023, we agreed to remove this change from this PR, as it will be covered in a second PR related to the voting process  @noatamir

## Evolution of PDEPs

> Most PDEPs are not expected to change after accepted. Once there is agreement in the changes,

Reviewer 3: ```suggestion
Most PDEPs are not expected to change after they are accepted. However, this PDEP (PDEP-1), which governs the PDEP process, may change as the process evolves.  Once there is agreement on the changes,
```

## Evolution of PDEPs

> Most PDEPs are not expected to change after they are accepted. However, this PDEP (PDEP-1), which governs the PDEP process, may change as the process evolves.  Once there is agreement on the changes,

Reviewer 5: ```suggestion
Most PDEPs are not expected to change after they are accepted. Once there is agreement on the changes,
```

This was already mentioned in the next paragraph.

to the `PDEP-N history` section. This will let readers understand that the PDEP has
changed and avoid confusion.

## <a id="List-of-issues"></a> List of issues that could have been PDEPs for context
### Clear examples for potential PDEPs:

- Adding a new parameter to many existing methods, or deprecating one in many places. For example:
  - The `numeric_only` deprecation affected many methods and could have been a PDEP.
- Adding a new data type has impact on a variety of places that need to handle the data type.
  Such wide-ranging impact would require a PDEP. For example:
  - `Categorical` ([GH-7217][7217], [GH-8074][8074]), `StringDtype` ([GH-8640][8640]), `ArrowDtype`
- A significant (breaking) change in existing behavior. For example:
  - Copy/view changes ([GH-36195][36195])
- Support of new python features with a wide impact on the project. For example:
  - Supporting typing within pandas vs. creation of `pandas-stubs` ([GH-43197][43197],
    [GH-45253][45253])
- New required dependency. For example:

>   - Moving rarely used I/O connectors to a separate repository [GH-28409](28409)

Reviewer 5: not sure this is an appropriate example for a new required dependency? (unless the assumption is we need pyarrow to do this?)

we probably should have an example that it more obvious without in-depth knowledge.

Reviewer 4: This is indeed not an example of a new required dependency. 
I would say it are just two independent examples of a case that could use a PDEP? (so not nesting this one under the previous line)

> - Significant changes to contributors' processes aren't going to have an impact on users, but 

Reviewer 5: ```suggestion
- Significant changes to contributors' processes are not going to have an impact on users, but 
```

  the do benefit from structured discussion among the contributors. For example:
  - Changing the build system to meson ([GH-49115][49115])

### Borderline examples:
Small changes to core functionality, such as `DataFrame` and `Series`, should always be 
considered as a PDEP candidate as it will likely have a big impact on users. But the same types 

> of changes in to functionalities would not be good PDEP candidates. That said, any discussion, 

Reviewer 5: not sure what this means? there is a grammatical error here?

of changes in other functionalities would not be good PDEP candidates. That said, any discussion,
no matter how small the change, which becomes controversial is a PDEP candidate. Consider if more
attention and/or a formal decision-making process would help. Following are some examples we
hope can help clarify our meaning here:

- API breaking changes, or discussion thereof, could be a PDEP. For example:
  - Value counts rename ([GH-49497][49497]). The scope does not justify a PDEP at first, but later a
    discussion about whether it should be executed as a breaking change or with deprecation
    emerges, which could benefit from a PDEP process.

> - Adding new parameters or methods to an existing method typically will not require a PDEP for
>   non-core features. For example:

Reviewer 7: ```suggestion
- Adding new methods or parameters to an existing method typically will not require a PDEP for
  non-core features. For example:
```

    wouldn't have required a PDEP.

>   - On the other hand, `DataFrame.assign()` might. While it's a single method without backwards

Reviewer 5: ```suggestion
  - On the other hand, `DataFrame.assign()` might. While it is a single method without backwards
```

>     compatibility concerns, it’s also a core feature and the discussion should be highly visible.

Reviewer 5: ```suggestion
    compatibility concerns, it is also a core feature and the discussion should be highly visible.
```

- Deprecating or removing a single method would not require a PDEP in most cases. For example:
  - `DataFrame.xs` ([GH-6249][6249]) is an example of deprecations on core features that would be

>     good a candidate for a PDEP.

Reviewer 5: ```suggestion
    a good candidate for a PDEP.
```

- New top level modules and/or exposing internal classes. For example:
  - Add `pandas.api.typing` ([GH-48577][48577]) is relatively small and wouldn’t necessarily
    require a PDEP.

### A counter-example:

- Significant changes to contributors' processes don't require a PDEP as they aren't going to
  have an impact on users. For example:
  - Changing the build system to meson ([GH-49115][49115])

### PDEP-1 History

- 3 August 2022: Initial version

> - 15 February 2023: Version 2

Reviewer 4: Do we want to point to this PR here?

Author: I can add that. 
I also wasn't sure if it should be the date the PR opened, or the date it was closed. WDYT

to the `PDEP-N history` section. This will let readers understand that the PDEP has
changed and avoid confusion.

## <a id="List-of-issues"></a> List of issues that could have been PDEPs for context
### Clear examples for potential PDEPs:

- Adding a new parameter to many existing methods, or deprecating one in many places. For example:
  - The `numeric_only` deprecation affected many methods and could have been a PDEP.
- Adding a new data type has impact on a variety of places that need to handle the data type.
  Such wide-ranging impact would require a PDEP. For example:
  - `Categorical` ([GH-7217][7217], [GH-8074][8074]), `StringDtype` ([GH-8640][8640]), `ArrowDtype`
- A significant (breaking) change in existing behavior. For example:
  - Copy/view changes ([GH-36195][36195])
- Support of new python features with a wide impact on the project. For example:
  - Supporting typing within pandas vs. creation of `pandas-stubs` ([GH-43197][43197],
    [GH-45253][45253])
- New required dependency. 

> - Removing module from the project or splitting it off to a seperate repository:

Reviewer 3: ```suggestion
- Removing module from the project or splitting it off to a separate repository:
```

  - Moving rarely used I/O connectors to a separate repository [GH-28409](28409)
- Significant changes to contributors' processes are not going to have an impact on users, but 

>   the do benefit from structured discussion among the contributors. For example:

Reviewer 3: ```suggestion
  they do benefit from structured discussion among the contributors. For example:
```

  - Changing the build system to meson ([GH-49115][49115])

### Borderline examples:
Small changes to core functionality, such as `DataFrame` and `Series`, should always be 
considered as a PDEP candidate as it will likely have a big impact on users. But the same types 
of changes in other functionalities would not be good PDEP candidates. That said, any discussion, 
no matter how small the change, which becomes controversial is a PDEP candidate. Consider if more 
attention and/or a formal decision-making process would help. Following are some examples we 
hope can help clarify our meaning here:

- API breaking changes, or discussion thereof, could be a PDEP. For example:
  - Value counts rename ([GH-49497][49497]). The scope does not justify a PDEP at first, but later a

>     discussion about whether it should be executed as breaking change or with deprecation

Reviewer 3: ```suggestion
    discussion about whether it should be executed as a breaking change or with deprecation
```

    emerges, which could benefit from a PDEP process.
- Adding new parameters or methods to an existing method typically will not require a PDEP for
  non-core features. For example:
  - Both `dropna(percentage)` ([GH-35299][35299]), and `Timestamp.normalize()` ([GH-8794][8794])

>     wouldn't have required a PDEP.

Reviewer 3: ```suggestion
    would not have required a PDEP.
```

to the `PDEP-N history` section. This will let readers understand that the PDEP has
changed and avoid confusion.

## <a id="List-of-issues"></a> List of issues that could have been PDEPs for context
### Clear examples for potential PDEPs:

- Adding a new parameter to many existing methods, or deprecating one in many places. For example:

>   - The `numeric_only` deprecation affected many methods and could have been a PDEP.

Reviewer 5: should this have a GH ref like most of the others?

- Adding a new data type has impact on a variety of places that need to handle the data type.
  Such wide-ranging impact would require a PDEP. For example:

>   - `Categorical` ([GH-7217][7217], [GH-8074][8074]), `StringDtype` ([GH-8640][8640]), `ArrowDtype`

Reviewer 5: same for `ArrowDtype`

Reviewer 4: I don't directly find a clear issue for ArrowDtype, only various PRs implementing parts of it.

Reviewer 5: no probs. not a blocker.

- A significant (breaking) change in existing behavior. For example:
  - Copy/view changes ([GH-36195][36195])

> - Support of new python features with a wide impact on the project. For example:

Reviewer 5: ```suggestion
- Support of new Python features with a wide impact on the project. For example:
```

  - Supporting typing within pandas vs. creation of `pandas-stubs` ([GH-43197][43197],
    [GH-45253][45253])
- New required dependency.
- Removing module from the project or splitting it off to a separate repository:
  - Moving rarely used I/O connectors to a separate repository [GH-28409](28409)
- Significant changes to contributors' processes are not going to have an impact on users, but
they do benefit from structured discussion among the contributors. For example:
  - Changing the build system to meson ([GH-49115][49115])

### Borderline examples:
Small changes to core functionality, such as `DataFrame` and `Series`, should always be
considered as a PDEP candidate as it will likely have a big impact on users. But the same types
of changes in other functionalities would not be good PDEP candidates. That said, any discussion,
no matter how small the change, which becomes controversial is a PDEP candidate. Consider if more
attention and/or a formal decision-making process would help. Following are some examples we
hope can help clarify our meaning here:

- API breaking changes, or discussion thereof, could be a PDEP. For example:

>   - Value counts rename ([GH-49497][49497]). The scope does not justify a PDEP at first, but later a

Reviewer 5: ```suggestion
  - `value_counts` rename ([GH-49497][49497]). The scope does not justify a PDEP at first, but later a
```

    discussion about whether it should be executed as a breaking change or with deprecation
    emerges, which could benefit from a PDEP process.
- Adding new methods or parameters to an existing method typically will not require a PDEP for
  non-core features. For example:
  - Both `dropna(percentage)` ([GH-35299][35299]), and `Timestamp.normalize()` ([GH-8794][8794])
    would not have required a PDEP.
  - On the other hand, `DataFrame.assign()` might. While it is a single method without backwards
    compatibility concerns, it is also a core feature and the discussion should be highly visible.
- Deprecating or removing a single method would not require a PDEP in most cases. For example:

>   - `DataFrame.xs` ([GH-6249][6249]) is an example of deprecations on core features that would be

Reviewer 5: this should maybe be a "On the other hand" case with an additional example that supports the "Deprecating or removing a single method would not require a PDEP" case. (to be consistent with the "Adding new methods or parameters to an existing method typically will not require a PDEP" examples above.)

Reviewer 4: Good point. I would maybe change the example to `DataFrame.append` deprecation, which I think was a clear example of a controversial deprecation (at least for users).

no matter how small the change, which becomes controversial is a PDEP candidate. Consider if more
attention and/or a formal decision-making process would help. Following are some examples we
hope can help clarify our meaning here:

- API breaking changes, or discussion thereof, could be a PDEP. For example:
  - `value_counts` result rename ([GH-49497][49497]). The scope does not justify a PDEP at first, but later a
    discussion about whether it should be executed as a breaking change or with deprecation
    emerges, which could benefit from the PDEP process.
- Adding new methods or parameters to an existing method typically will not require a PDEP for
  non-core features. For example:
  - Both `dropna(percentage)` ([GH-35299][35299]), and `Timestamp.normalize()` ([GH-8794][8794])
    would not have required a PDEP.
  - On the other hand, `DataFrame.assign()` might. While it is a single method without backwards
    compatibility concerns, it is also a core feature and the discussion should be highly visible.
- Deprecating or removing a single method would not require a PDEP in most cases. For example:

>   - That said, `DataFrame.append` ([GH-35407][35407]) is an example of deprecations on core

Reviewer 5: perhaps also need to remove the `For example:` to make this scan better if not adding a case where a PDEP is not required.

to the `PDEP-N history` section. This will let readers understand that the PDEP has
changed and avoid confusion.

## <a id="List-of-issues"></a> List of issues that could have been PDEPs for context
### Clear examples for potential PDEPs:
- Adding a new parameter to many existing methods, or deprecating one in many places. For example:

>   - The numeric_only deprecation affected many methods and could have been a PDEP.

Reviewer 3: put backquotes around `numeric_only`

- Adding a new data type has impact on a variety of places that need to handle the data type.
  Such wide-ranging impact would require a PDEP. For example:
  - Categorical ([GH-7217][7217], [GH-8074][8074]), StringDtype ([GH-8640][8640]), ArrowDtype
- A significant (breaking) change in existing behavior. For example:
  - copy/view changes ([GH-36195][36195])
- Support of new python features with a wide impact on the project. For example:
  - Supporting typing within pandas vs. creation of pandas-stubs ([GH-43197][43197],

>     [GH-45253][45253]) New required dependency

Reviewer 3: Not sure that "New required dependency" makes sense in this context.  I think we want that as a separate bullet.

### Borderline examples:

> - Small changes to core functionality, such as DataFrame and Series, should always be considered

Reviewer 3: backquotes around `DataFrame` and `Series`

  changes in other methods would not be good PDEP candidates. That said, any discussion, no
  matter how small the change, which becomes controversial is a PDEP candidate. Consider if more
  attention and/or a formal decision-making process would help. Following are some examples we
  hope can help clarify our meaning here:
- API breaking changes, or discussion thereof, could be a PDEP. For example:
  - Value counts rename ([GH-49497][49497]). The scope doesn’t justify a PDEP at first, but later a
    discussion about whether it should be executed as breaking change or with deprecation
    emerges, which could benefit from a PDEP process.
- Adding new parameters or methods to an existing method typically won't require a PDEP for
  non-core features. For example:
  - Both dropna(percentage) ([GH-35299][35299]), and Timestamp.normalize() ([GH-8794][8794])
    wouldn't have required a PDEP.
  - On the other hand, DataFrame.assign() might. While it's a single method without backwards
    compatibility concerns, it’s also a core feature and the discussion should be highly visible.
- Deprecating or removing a single method would not require a PDEP in most cases. For example:

>   - DataFrame.xs ([GH-6249][6249]) is an example of depredations on core features that would be

Reviewer 3: "deprecations" not "depredations"

    good a candidate for a PDEP.
- Changing the default value of parameters in a core pandas method is another edge case. For
  example:

>   - Such changes in dropna,  DataFrame.groupby, or in Series.groupby could be PDEPs.

Reviewer 3: add appropriate backquotes

- New top level modules and/or exposing internal classes. For example:
  - Add pandas.api.typing ([GH-48577][48577]) is relatively small and wouldn’t necessarily
    require a PDEP.

### A counter-example:
- Significant changes to contributors' processes don't require a PDEP as they aren't going to
  have an impact on users. For example:

>   - Changing the build system to meson

Reviewer 3: can refer to https://github.com/pandas-dev/pandas/pull/49115 here

to the `PDEP-N history` section. This will let readers understand that the PDEP has
changed and avoid confusion.

## <a id="List-of-issues"></a> List of issues that could have been PDEPs for context
### Clear examples for potential PDEPs:

> - Adding a new parameter to many existing methods, or deprecating one in many places. For example:

Reviewer: Some markdown parsers require a blank line before lists. Ours doesn't, but for other times maybe good to leave it in case we ever change to a more strict one.

  - The `numeric_only` deprecation affected many methods and could have been a PDEP.
- Adding a new data type has impact on a variety of places that need to handle the data type.
  Such wide-ranging impact would require a PDEP. For example:
  - `Categorical` ([GH-7217][7217], [GH-8074][8074]), `StringDtype` ([GH-8640][8640]), `ArrowDtype`
- A significant (breaking) change in existing behavior. For example:
  - Copy/view changes ([GH-36195][36195])
- Support of new python features with a wide impact on the project. For example:
  - Supporting typing within pandas vs. creation of `pandas-stubs` ([GH-43197][43197],
    [GH-45253][45253])

> - New required dependency

Reviewer 6: Maybe https://github.com/pandas-dev/pandas/issues/28409 could be a good example of splitting out a module in addition to the typing one.

Author: Thanks!

### Borderline examples:

> - Small changes to core functionality, such as `DataFrame` and `Series`, should always be considered

Reviewer 4: This first bullet point should maybe be a normal paragraph, and only the remaining bullet points as a list? (this bullet point also ends with a `:`, so I would expect the other points to be a sublist of this one, otherwise)

  as a PDEP candidate as it will likely have a big impact on users. But the same types of

>   changes in other methods would not be good PDEP candidates. That said, any discussion, no

Reviewer 4: ```suggestion
  changes in other objects would not be good PDEP candidates. That said, any discussion, no
```

? (not fully sure what our intent was here, but I think the idea was that for example adding a new core method to DataFrame has wider impact than a new method on eg Timestamp)

  non-core features. For example:
  - Both `dropna(percentage)` ([GH-35299][35299]), and `Timestamp.normalize()` ([GH-8794][8794])
    wouldn't have required a PDEP.
  - On the other hand, `DataFrame.assign()` might. While it's a single method without backwards
    compatibility concerns, it’s also a core feature and the discussion should be highly visible.
- Deprecating or removing a single method would not require a PDEP in most cases. For example:
  - `DataFrame.xs` ([GH-6249][6249]) is an example of deprecations on core features that would be
    good a candidate for a PDEP.
- Changing the default value of parameters in a core pandas method is another edge case. For
  example:
  - Such changes in `dropna`,  `DataFrame.groupby`, or in `Series.groupby` could be PDEPs.
- New top level modules and/or exposing internal classes. For example:
  - Add `pandas.api.typing` ([GH-48577][48577]) is relatively small and wouldn’t necessarily
    require a PDEP.

### A counter-example:
- Significant changes to contributors' processes don't require a PDEP as they aren't going to
  have an impact on users. For example:

>   - Changing the build system to meson ([GH-49115][49115])

Reviewer: We did have a PDEP for that, and I think it was quite useful. What do you think @WillAyd

Reviewer 8: Yea I think this was worth a PDEP - it definitely falls under the category of a significat `internal change`

Author: thanks! 🙇

Author: Would you say such process changes would be boreder-line cases or clear PDEP examples? I'm wondering where to move this example too

Reviewer 8: I think this one is clear. If it's a significant change to the developer workflow I think it could benefit from a PDEP whether user facing or not. But no hard opinion it others disagree

### PDEP-1 History

- 3 August 2022: Initial version

> - 15 February 2023: Revision 1

Reviewer: I think we're using revision 1 for the initial revision, should this be revision 2?

Reviewer 5: a summary of the changes could help
